python/ql/lib/semmle/python/frameworks/Pyramid.qll

```diff
@@ -0,0 +1,87 @@
+/**
+ * Provides classes modeling security-relevant aspects of the `pyramid` PyPI package.
+ * See https://docs.pylonsproject.org/projects/pyramid/.
+ */
+
+private import python
+private import semmle.python.dataflow.new.DataFlow
+private import semmle.python.dataflow.new.RemoteFlowSources
+private import semmle.python.dataflow.new.TaintTracking
+private import semmle.python.Concepts
+private import semmle.python.ApiGraphs
+private import semmle.python.dataflow.new.FlowSummary
+private import semmle.python.frameworks.internal.PoorMansFunctionResolution
+private import semmle.python.frameworks.data.ModelsAsData
+
+/**
+ * Provides models for the `pyramid` PyPI package.
+ * See https://docs.pylonsproject.org/projects/pyramid/.
+ */
+module Pyramid {
+  // TODO: qldoc
+  module View {
+    /**
+     * A callable that could be used as a pyramid view callable.
+     */
+    private class PotentialViewCallable extends Function {
+      PotentialViewCallable() {
+        this.getPositionalParameterCount() = 1 and
+        this.getArgName(0) = "request"
+        or
+        this.getPositionalParameterCount() = 2 and
+        this.getArgName(0) = "context" and
+        this.getArgName(1) = "request"
+      }
+
+      /** Gets the `request` parameter of this view callable. */
+      Parameter getRequestParameter() { result = this.getArgByName("request") }
+    }
+
+    abstract class ViewCallable extends PotentialViewCallable, Http::Server::RequestHandler::Range {
+      override Parameter getARoutedParameter() { result = this.getRequestParameter() }
```

Comment:
for a route like `/hello/<name>` that is handled by `def handle_hello(request, name): ...` the `<name>` is a routed parameter. I haven't looked at the pyramid docs whether this is something they do, but that's sort of what it's for.

Reply:
I don't think pyramid has parameters like that (instead `name` would be in a `matchdict` object on the `request`.
Should the `request` parameter be modeled as a remote flow source separately from this then?

Reply:
> Should the request parameter be modeled as a remote flow source separately from this then?

Yes please :+1: